### ISSUE TITLE
Coloring stdout red restricts usefulness

### ISSUE THREAD

**Issue by EvanDorsky** [wting/autojump#377]
Autojump pipes the directory that it chooses to stdout, which is great. But because it makes the text color red, I can't do something like this:

    $ mv doc.md `j Markdown`

If I know where `j Markdown` is going to go, this is extremely useful. I can compose with autojump. But this doesn't work by default because the special characters surrounding the path to make it red keep shell scripts from seeing the output of autojump as a directory.

I don't see the overwhelming advantage of coloring stdout red. In the meantime, if you want this functionality, you can use sed to strip the color characters:

    function jj() {
        j $* | sed -r "s/\x1B\[([0-9]{1,2}(;[0-9]{1,2})?)?[m|K]//g" # not my original pattern
    }

But autojump could be a better Unix citizen if it just piped the directory, unadorned, to stdout. This change was easy to make in the zsh source, but I don't know all the supported shells well enough to make a pull request.

**Comment by wting** [comment]
`j` is just a wrapper around `autojump`. Does `mv doc.md $(autojump Markdown)` work for you?

**Comment by EvanDorsky** [comment]
Yes, it does — thanks for clarifying. I guess I don't see a reason for the shorter `j` to remove this functionality, though.

**Comment by wting** [comment]
So we're trying to separate two things here. `autojump` is the thing that works with input / output to find the best response. `j` is a thin wrapper around `autojump`. `j` isn't removing functionality, rather it's adding functionality for command line use.

**Comment by EvanDorsky** [comment]
I think I understand the distinction. I'm just trying to point out that when using `$( )`, `autojump` works while `j` doesn't.

**Comment by blueyed** [comment]
@EvanDorsky 
`j` is not meant to be used with `$(…)`, but `j` is.

OTOH `j` could maybe detect that it is being used like that and then skip the coloring?!

**Comment by EvanDorsky** [comment]
@blueyed
>`j` is not meant to be used with $(…), but `j` is.

I don't think I'm following you — which of the two `j`s in the quoted sentence was supposed to be something else?

**Comment by wting** [comment]
Pretty sure @blueyed  meant this:

> `j` is not meant to be used with $(…), but **`autojump`** is.

**Comment by blueyed** [comment]
Yeah, sorry for the confusion.

**Comment by EvanDorsky** [comment]
Okay. I guess there's a tradeoff in functionality here between `j` printing in red and being able to use `j` with `$( )` that I don't fully understand. Could you shed some light on that decision? If you feel that everything is working as intended, feel free to close this issue. Thanks.

**Comment by kuerzn** [comment]
Without a check for TTY the red color does more harm than good. Auto detecting the tty like other GNU tools (grep, ls, ..) do is an unwritten law in UNIX. Without is one of the key features, namely composability, is destroyed.

**Comment by joelthelion** [comment]
The red color is important because the destination directory is unpredictable. So it's important to draw the user's attention on where he ended up.

That said, maybe `j` could detect whether the output is a TTY.

**Comment by lilydjwg** [comment]
For me the red text isn't as noticeable as white one on my dark terminal...
2016年2月2日 下午7:15，"Joel Schaerer" <notifications@github.com>写道：

**Comment by wting** [comment]
Fixed in `release-v22.3.0` from @johannesgerer's patch.

**Comment by eexpress** [comment]
I think export an environment parameter for the output ansi color in  /usr/share/autojump/autojump.bash is a good idea. So we can change the color follow ". /usr/share/autojump/autojump.bash" in .bashrc or etc.

just like this:
```
. /usr/share/autojump/autojump.bash
export AJoutputcolor="\[\e[1;37;42m"   <- ansi or tput here
```
